ops/lib.rs

```diff
@@ -111,24 +111,68 @@ impl Op {
       active,
     } = fast_call::generate(&core, &mut optimizer, &item);
 
+    let docline = format!("Use `{name}::decl()` to get an op-declaration");
+
+    let is_v8 = attrs.is_v8;
+    let is_unstable = attrs.is_unstable;
+
+    if let Some(v8_fn) = attrs.relation {
+      return quote! {
+        #[allow(non_camel_case_types)]
+        #[doc="Auto-generated by `deno_ops`, i.e: `#[op]`"]
+        #[doc=""]
+        #[doc=#docline]
+        #[doc="you can include in a `deno_core::Extension`."]
+        pub struct #name;
+
+        #[doc(hidden)]
+        impl #name {
+          pub fn name() -> &'static str {
+            stringify!(#name)
+          }
+
+          pub fn v8_fn_ptr #generics () -> #core::v8::FunctionCallback #where_clause {
+            use #core::v8::MapFnTo;
+            #v8_fn::v8_func::<#type_params>.map_fn_to()
+          }
+
+          pub fn decl #generics () -> #core::OpDecl #where_clause {
+            #core::OpDecl {
+              name: Self::name(),
+              v8_fn_ptr: Self::v8_fn_ptr::<#type_params>(),
+              enabled: true,
+              fast_fn: #decl,
+              is_async: #is_async,
+              is_unstable: #is_unstable,
+              is_v8: #is_v8,
+              argc: 0,
+            }
+          }
+
+          #[inline]
+          #[allow(clippy::too_many_arguments)]
+          #orig
+        }
+
+        #impl_and_fn
+      };
+    }
+
     let has_fallible_fast_call = active && optimizer.returns_result;
 
     let (v8_body, argc) = if is_async {
+      let deferred = attrs.deferred;
```

Comment:
Is this extraction necessary?

Reply:
yes, `Attributes` doesn't impl Copy anymore